com.unity.shadergraph/Editor/Data/Graphs/SamplerStateShaderProperty.cs

```diff
@@ -46,7 +46,7 @@ internal override string GetHLSLVariableName(bool isSubgraphProperty)
             if (isSubgraphProperty)
                 return referenceName;
             else
-                return $"UnityBuildSamplerStateStruct({referenceName})";
+                return $"UnityBuildSamplerStateStruct({propertyReferenceName})";
```

Comment:
subgraphs don't refer to the HLSL property, they refer to the local subgraph function input parameter.  So they use the reference name.